octavia_f5/restclient/as3classes.py

```diff
@@ -305,3 +305,11 @@ class HTTP_Profile(BaseDescription):
     def __init__(self, **kwargs):
         super(HTTP_Profile, self).__init__(locals())
         setattr(self, 'class', 'HTTP_Profile')
+
+
+class Data_Group(BaseDescription):
+    def __init__(self, _records, keyDataType='ip', **kwargs):
+        super(Data_Group, self).__init__(locals())
+        setattr(self, 'class', 'Data_Group')
+        self.require('keyDataType')
+        setattr(self, 'records', [{"key": r} for r in _records])
```

Comment:
The [Data_Group_Records AS3 class](https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/refguide/schema-reference.html#data-group-records) has the attribute `value`, not `key`. This seems to be an error in the example AS3 declaration provided in the Jira ticket as well. I've asked @m-kratochvil, but he hasn't responded yet, so I'll send this review for now.

Reply:
The example is correct. If `"keyDataType": "ip"` then it means the `key` is the actual `ip` and the `value` is optional, if used, it serves more like a description for the ip. In the AS3 schema reference, you have to look at [Data_Group_Records_IP](https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/refguide/schema-reference.html#data-group-records-ip) instead. You can also see this in the actual AS3 schema [HERE](https://github.com/F5Networks/f5-appsvcs-extension/blob/b80bc1d7e1a3d2c0f9c5e4d05dee7d25a3b726d6/schema/3.36.0/as3-schema.json#L15818-L15828).

The way IP objects can be defined in the Big-IP TMOS can be confusing, as it allows the name of the object to be the actual value. Data groups adopted this method of defining IP addresses.

So this is perfectly correct data group definition in AS3:
```
            "test_datagroup": {
                "class": "Data_Group",
                "keyDataType": "ip",
                "records": [
                    {
                        "key": "10.1.0.0/24",
                        "value": ""
                    },
                    {
                        "key": "10.10.0.0/16",
                        "value": "awesome_prefix"
                    },
                    {
                        "key": "10.10.1.1"
                    }
                ]
            }
        }
    }
```
And for better understanding, here is the same data group in TMOS:
```
ltm data-group internal test_datagroup {
    records {
        10.1.0.0/24 { }
        10.10.0.0/16 {
             awesome_prefix
         }
        10.10.1.1 { }
    }
    type ip
}
```

Reply:
Okay wow, I see that one has to be careful with these. Thanks for the explanation!